### PR TITLE
Output notx 6846 v7.8

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6973

Describe changes:
- detect: log relevant frames app-layer metdata
- frames: change rust API to make tx_id mandatory
- frames: set right tx_id for SIP and websocket and DNS

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1801

https://github.com/OISF/suricata/pull/10979 with nit doc fix

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20414

File: rust/src/dns/dns.rs [hunk 1]
```diff
@@ -381,7 +381,7 @@ impl DNSState {
         None
     }
 
-    fn parse_request(&mut self, input: &[u8], is_tcp: bool) -> bool {
+    fn parse_request(&mut self, input: &[u8], is_tcp: bool, fid: Option<Frame>, flow: *const core::Flow,) -> bool {
```
Comment (jufajardini) [inline rust/src/dns/dns.rs:384]: This is even more nit than the other comment, and not a blocker. Why `fid` and not `frame`?

if this isn't something worth changing, all looks good to me :)

Comment (catenacyber) [reply rust/src/dns/dns.rs:384]: Because I thought first I would only use the frame id...
Changing that

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/10995

**Pull request closed without merging**